File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.ResponseResult;
import com.pinyougou.pojo.TbBrand;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface BrandService {
    /** 
    * @Description: 查找到所有的品牌消息
    * @Param: [] 
    * @return: java.util.List<com.pinyougou.pojo.TbBrand> 
    * @Author: Yin 
    * @Date: 2018/8/29 
    */ 
    List<TbBrand> findAll();

    /** 
    * @Description: 分页查询所有的 
    * @Param: [page, size] 
    * @return: java.util.List<com.pinyougou.pojo.TbBrand> 
    * @Author: Yin 
    * @Date: 2018/8/29 
    */
    Map<String, Object> findByPage(Integer page, Integer size);

    /** 
    * @Description: 添加品牌
    * @Param: [brand] 
    * @return: com.pinyougou.entity.ResponseResult<com.pinyougou.pojo.TbBrand> 
    * @Author: Yin 
    * @Date: 2018/8/29 
    */ 
    ResponseResult<TbBrand> addBrand(TbBrand brand);

    /** 
    * @Description: 对品牌进行修改
    * @Param: [brand] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/8/29 
    */ 
    void updateBrand(TbBrand brand);

    /** 
    * @Description: 根据id去寻找对应的品牌
    * @Param: [id] 
    * @return: com.pinyougou.pojo.TbBrand 
    * @Author: Yin 
    * @Date: 2018/8/29 
    */ 
    TbBrand findById(Integer id);

    /** 
    * @Description: 批量删除品牌信息
    * @Param: [ids] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/8/30 
    */ 
    void deleteBrand(String ids);

    /** 
    * @Description: 条件查询品牌的信息 
    * @Param: [brand, page, rows] 
    * @return: java.util.List<com.pinyougou.pojo.TbBrand> 
    * @Author: Yin 
    * @Date: 2018/8/30 
    */ 
    Map<String, Object> findByCondiction(TbBrand brand, Integer page, Integer rows);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//这是品牌的服务层
app.service('brandService',function ($http) {
    /*
    * 查找所有，也就是不进行分页也不进行条件查询
    * */
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }
    /*
    * 条件查询，其实也就是分页的情况
    * */
    this.findByCondition=function (page,rows,entity) {
        return  $http.post("../brand/findByCondition.do?page="+page+"&rows="+rows,entity);
    }
    /*
    * 增加品牌的信息
    * 若是存在了id的情况，那就改成修改的作用
    * */
    this.addBrand=function (entity) {
        var method = 'add';
        if (entity.id!=null){
            method = 'update';
        }
        return $http.post("../brand/"+method+".do",entity);
    }
    /*
    * 根据id查找到信息，也就是修改前的回显
    * */
    this.findById = function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }
    /*
    * 删除品牌
    * */
    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }

    /*
    * 模板上所需要的查询
    * */
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }


})